src/main/java/nl/knaw/dans/migration/core/DataverseLoader.java

```diff
@@ -116,16 +97,40 @@ public void loadFromDataset(String doi) {
             actualDataset.setDepositor(depositor);
             actualDataset.setFileAccessRequest(lastVersion.isFileAccessRequest());
             actualDataset.setCitationYear(publicationDate.substring(0,4));
-            saveActualDataset(actualDataset);
+            loadFiles(shortDoi, v, actualDataset);
         }
     }
 
-    private ActualFile toActual(FileMeta fileMeta, String doi, int majorVersion, int minorVersion, boolean datasetHasAccessRequestEnabled) {
+    private void loadFiles(String doi, DatasetVersion v, ActualDataset actualDataset) {
```

Comment:
Getting a `No session currently bound to execution context` when moving the `UnitOfWork` annotation to this method. The exception occurs on the first attempt to insert anything, even when in lining the save methods and/or saving the actual-dataset first.

Reply:
Putting `UnitOfWork` at the caller (`loadFromDataset`) did work, on this method -even when made public- did not work.